lib/train-habitat/httpgateway.rb

```diff
@@ -15,6 +17,23 @@ def initialize(host)
       def services
         JSON.parse(Net::HTTP.get_response(uri).body)
       end
+
+      def service(origin, name)
+        service = services.select do |svc|
+          svc.dig('pkg', 'origin') == origin && svc.dig('pkg', 'name') == name
+        end
```

Comment:
Hrm, how would you feel about...
```ruby
def service(origin, name)
  by_origin = ->(s) { s.dig('pkg', 'origin') == origin }
  by_name   = ->(s) { s.dig('pkg', 'name') == name }

  matching_services = services.select(&by_origin).select(&by_name)

  validate!(matching_services, origin, name) or service.first
end
```

Reply:
I pulled out the first two lines to methods to try to keep the method at the scope. Let me know what you think.